### PR TITLE
Store app selection counts internally and use them to sort results

### PR DESCRIPTION
**Pull request by TheConnMan**
With a couple small changes I've gotten it so search results sort based on selection count, but only within the current session. I could use some help persisting these counts to the `CACHE_FILE`, which as I understand it only gets populated by an async thread, so there's no internal API for saving the cache file.

Ref #91

### CODE REVIEW

File: app/main/containers/Search/index.js [hunk 1]
```diff
@@ -369,7 +369,7 @@ Search.propTypes = {
 function mapStateToProps(state) {
   return {
     selected: state.search.selected,
-    results: state.search.resultIds.map(id => state.search.resultsById[id]),
+    results: state.search.resultIds.map(id => state.search.resultsById[id]).sort((a, b) => b.selectCount - a.selectCount),
```
Comment (KELiON) [inline app/main/containers/Search/index.js]: This sort will be applied for all results, not only from `basic-apps`, so would be better to move it to plugin

File: app/main/plugins/core/basic-apps/index.js [hunk 2]
```diff
@@ -27,7 +28,10 @@ const fn = ({ term, actions, display }) => {
           event.preventDefault()
         }
       },
-      onSelect: () => openApp(app),
+      onSelect: () => {
+        app.selectCount = app.selectCount ? app.selectCount + 1 : 1;
```
Comment (KELiON) [inline app/main/plugins/core/basic-apps/index.js]: So, you can use the same logic, but move sorting to this file [before](https://github.com/TheConnMan/cerebro/blob/2d590903c15ceaf860945a0e760f773068623937/app/main/plugins/core/basic-apps/index.js#L14) `map`

File: app/main/plugins/core/basic-apps/index.js [hunk 3]
```diff
@@ -16,6 +16,7 @@ const fn = ({ term, actions, display }) => {
     return {
       icon,
       id: id || path,
+      count: app.selectCount || 0,
```
Comment (KELiON) [inline app/main/plugins/core/basic-apps/index.js]: `count` key will be just ignored. I'd not add new key for public plugin API to only sort results.

**Comment by maximbaz** [comment]
Thanks for taking up this feature request :+1: 

Yeah, persisting the counter in `CACHE_FILE` is essential for this feature, @KELiON do you have any hints how this can be achieved?

File: app/main/containers/Search/index.js (hunk 1 above)
Comment (TheConnMan) [inline app/main/containers/Search/index.js]: I figured out the issue I was having when sorting only the basic apps. All apps start out with NaN as their selection count, so my sort wasn't working out. My pushed changes isolate the sorting to only basic apps.

**Pull request merged**